observatory-platform/observatory/platform/workflows/stream_telescope.py

```diff
@@ -333,14 +326,28 @@ def bq_append_new(self, release: StreamRelease, **kwargs):
                 partition_table_id,
             )
 
+    def bq_create_snapshot(self, release: StreamRelease, **kwargs):
+        """ Create a snapshot from an existing BigQuery table
+
+        :param release: a StreamRelease instance
+        :param kwargs: The context passed from the PythonOperator.
+        :return: None.
+        """
+        logging.info("Creating a BigQuery snapshot")
+        bq_load_info = self.get_bq_load_info(release)
+        for _, main_table_id, _ in bq_load_info:
+            destination_table_id = bigquery_sharded_table_id(main_table_id + "_snapshots", release.end_date)
+            create_bigquery_snapshot(self.dataset_id, main_table_id, self.dataset_id, destination_table_id)
```

Comment:
Perhaps not for this PR as it will affect many other workflows, but it would be cleaner to explicitly pass in a project id.